Parser/Token.cs

```diff
@@ -38,10 +39,17 @@ public enum E {
    // Utility function used to echo an error to the console
    public void PrintError () {
       if (Kind != ERROR) throw new Exception ("PrintError called on a non-error token");
+      int start = Line > 3 ? Line - 3 : 0, end = Line + 2 <= Source.Lines.Length ? Line + 2 : Source.Lines.Length, line = start + 1;
+      Console.WriteLine ($"File: {Source.FileName}\n───┬────────────────");
+      foreach (var str in Source.Lines[start..Line]) Console.WriteLine ($"{line++,3}│ {str}");
+      mX = Source.Lines[Line - 1].IndexOf ($"{Text}");
       Console.ForegroundColor = ConsoleColor.Yellow;
-      Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
+      Console.WriteLine ("^", Console.CursorLeft = mX + 5);
+      Console.WriteLine ($"{ErrorMessage}", Console.CursorLeft = Math.Max (mX, Column - ErrorMessage.Length / 2) + 1);
       Console.ResetColor ();
+      Console.WriteLine ($"{line++,3}│ {string.Join ($"\n{line++,3}│ ", Source.Lines[Line..end])}");

```

Comment:
The second line after the error line doesn't exist. So, the number is line number is not required.
<img width="278" alt="image" src="https://github.com/Dakshita-Minocha/PSI-A18/assets/75829870/32fff568-4129-4a66-950c-87c71b496828">
The second line after the error line is not printed:
<img width="194" alt="image" src="https://github.com/Dakshita-Minocha/PSI-A18/assets/75829870/ff94eb5e-9bb5-4031-aa20-3b2adc375389">

Reply:
Should I just skip over it?

Reply:
@Dakshita-Minocha : For the first one I meant that the line number (12) is printed for the last line even if there is no text after the error. For the second one, next for the last number (4) is not printed. I think minor boundary condition issue.

Reply:
That's because that line is empty in the input text except for '\n' I think. 
I meant to ask if I should just directly print the next line instead?

Reply:
OK, if its an empty line leave it and don't print the next line

Reply:
Okay